app/lib/person_gross_income_subtotals.rb

```diff
@@ -0,0 +1,27 @@
+class PersonGrossIncomeSubtotals
+  def initialize(gross_income_components = {})
+    @total_gross_income = gross_income_components[:total_gross_income]
+    @monthly_student_loan = gross_income_components[:monthly_student_loan]
+    @monthly_unspecified_source = gross_income_components[:monthly_unspecified_source]
+    @regular_income_categories = gross_income_components[:regular_income_categories]
+    @employment_income_subtotals = gross_income_components[:employment_income_subtotals] || instantiate_blank_income_subtotals
```

Comment:
This class feels a bit annoying, as it does 1 useful thing with the data (the monthly regular incomes method) but otherwise just propagates nil values if it is instantiated by the default constructor. Is there some way of converting this into a full 'Null Object' pattern ?